plugins/msg-broker/mcollective/lib/openshift/mcollective_application_container_proxy.rb

```diff
@@ -1815,7 +1815,12 @@ def move_gear_post(gear, destination_container, state_map)
       if gear.group_instance.platform.downcase == "windows"
         log_debug "DEBUG: Restoring ownership and user ACLs for Windows gear '#{gear.uuid}'"
         rsync_keyfile = Rails.configuration.auth[:rsync_keyfile]
-        log_debug `eval \`ssh-agent\`; ssh-add #{rsync_keyfile}; ssh -o StrictHostKeyChecking=no -A root@#{destination_container.get_ip_address} "/cygdrive/c/openshift/bin/oo-cmd.exe oo-admin-restore-acls --uuid:#{gear.uuid}"; exit_code=$?; ssh-agent -k;exit $exit_code`
+        destination_address = case Rails.configuration.msg_broker[:node_rsync_address]
+          when 'public_hostname' then destination_container.get_public_hostname
+          when 'server_identity' then destination_container.id
+          else destination_container.get_ip_address
+        end
```

Comment:
por favor, factor these out into a method call... the only thing different is the container var :)